rotary_utils/rotate.py

```diff
@@ -0,0 +1,286 @@
+#!/usr/bin/env python
+# rotate.py
+# Rotates circular DNA elements
+# Copyright Jackson M. Tsuji and Lee H. Bergstrand 2024
+
+import argparse
+import logging
+import os
+import sys
+import math
+
+from Bio import SeqIO
+
+from rotary_utils.utils import set_write_mode, load_fasta_sequences
+
+logger = logging.getLogger(__name__)
+
+
+# TODO - this is just a placeholder function for now to show some of the features I'd like to add to the CLI later on.
+def main(args):
+    """
+    Starts the rotate utility
+    :param args: args parsed by parse_cli()
+    """
+
+    # Startup checks
+    if args.verbose is True:
+        logger.setLevel(logging.DEBUG)
+        logging.getLogger('rotary_utils.utils').setLevel(logging.DEBUG)
+    else:
+        logger.setLevel(logging.INFO)
+        logging.getLogger('rotary_utils.utils').setLevel(logging.INFO)
+
+    # Check output file
+    # TODO - add overwrite flag
+    # TODO - make this info a function and check all other output files
+    # (TODO - add prefix flag)
+    # TODO - consider also checking if the directory where the output file will be saved exists and is writeable
+    # TODO - include a warning like logger.warning(f'Output file already exists: "{args.output_fasta}". Files may be overwritten.') to the log
+    output_file_exists = os.path.isfile(args.output_fasta)
+
+    if (output_file_exists is True) & (args.overwrite is False):
+
+        logger.error(f'Output file already exists: "{args.output_fasta}". Will not continue. Set the '
+                     f'--overwrite flag at your own risk if you want to overwrite files.')
+        sys.exit(1)
+
+    # Parse some command line inputs further
+    # TODO - improve error handling if a string is provided instead of a real length
+    # TODO - confirm what the default value of args.contig_names will be - might need to specify
+    if args.contig_names is not None:
+        contig_names = [int(x) for x in args.contig_names.split(',')]
+
+    # Startup messages
+    logger.info('Running ' + os.path.basename(sys.argv[0]))
+    logger.debug('### SETTINGS ###')
+    logger.debug(f'Input FastA: {args.input_fasta}')
+    logger.debug(f'Output FastA: {args.output_fasta}')
+    # TODO - finish the rest of the startup messages
+
+    logger.debug(f'Threads: {args.threads}')
+    logger.debug(f'Verbose logging: {args.verbose}')
+    logger.debug('################')
+
+    # TODO - add rotate code
+
+    logger.info(os.path.basename(sys.argv[0]) + ': done.')
+
+
+def parse_gene_positions():
+    """
+    Parses the positions of genes along a contig
+    :return:
+    """
+    # TODO
```

Comment:
@jmtsuji When you leave method stubs like this, it is customary to add a pass call. See: https://stackoverflow.com/questions/7620382/stubbing-out-functions-or-classes

```python
def parse_gene_positions():
    """
    Parses the positions of genes along a contig
    :return:
    """
    # TODO
    pass
```

Reply:
@LeeBergstrand So sorry -- this `rotate.py` module is actually not yet ready for review and was not supposed to be in this PR. I recently merged changes from this branch into the development branch for `rotate` in PR #3, but because there was a merge conflict, it looks like `rotate` may have gotten copied into thisbranch when the conflict was resolved...

Reply:
These functions will get fleshed out soon, but because they've ended up in this PR, I'll go ahead and add a pass call  as you suggested.